nest-tax-backend/src/tax/tax.service.ts

```diff
@@ -209,4 +208,83 @@ export class TaxService {
       )
     }
   }
+
+  async getTaxDetail(
+    birthNumber: string,
+    year: number,
+  ): Promise<ResponseTaxSummaryDetailDto> {
+    const today = dayjs().tz('Europe/Bratislava')
+
+    const tax = await this.fetchTaxData(birthNumber, year)
+
+    const overallPaidTax = await this.getAmountAlreadyPaidByTaxId(tax.id)
```

Comment:
Why don't we include `taxPayments` inside `fetchTaxData`? We don't have to force reusability withing the old code and new code. 

I would include `taxPayments` and offload the logic to the pure function (I think it belongs there).